nodejs/examples/webhook-server/index.js

```diff
@@ -0,0 +1,72 @@
+const tlSigning = require("truelayer-signing");
+const axios = require('axios');
+const express = require('express');
+const app = express();
+
+const allowedJkus = {
+  "https://webhooks.truelayer.com/.well-known/jwks": true,
+  "https://webhooks.truelayer-sandbox.com/.well-known/jwks": true,
+};
+
+// global JWK cache
+const cachedJwks = {};
+async function get_jwks(sig) {
+  let kid = tlSigning.extractKid(sig);
+  tlSigning.SignatureError.ensure(kid, `Tl-Signature has missing key id`);
+
+  let jku = tlSigning.extractJku(sig);
+  tlSigning.SignatureError.ensure(allowedJkus[jku], `Tl-Signature has invalid jku: ${jku}`);
+
+  // check if we have this KID/JKU pair stored
+  let jwks = cachedJwks[jku];
+  if (jwks) {
+    for (let i = 0; i < jwks.keys.length; i++) {
+      if (jwks.keys[i].kid == kid) {
+        return jwks;
+      }
+    }
+  }
+
+  // otherwise, fetch the JWKs from the server
+  cachedJwks[jku] = (await axios.get(jku)).data;
+  return cachedJwks[jku];
+}
+
+async function verify_hook(req) {
+  // extract the Tl-Signature from the headers
+  let sig = req.headers["tl-signature"];
+  tlSigning.SignatureError.ensure(sig, "missing Tl-Signature header");
+
+  // get the (cached) JWKs for this request
+  let jwks = await get_jwks(sig);
+
+  // verify the request (will throw on failure)
+  tlSigning.verify({
+    signature: sig,
+    method: req.method,
+    path: req.path,
+    headers: req.headers,
+    body: JSON.stringify(req.body),
+    jwks: JSON.stringify(jwks),
+  });
+}
+
+app.post('/hook/:id',
```

Comment:
I think it's better to have this as the literal. In reality the webhook path is a constant, it's just the example webhook signature uses a uuid distinct path.

```suggestion
// Note: Webhook path can be whatever is configured, here a unique path 
// is used matching the README example signature.
app.post('/hook/d7a2c49d-110a-4ed2-a07d-8fdb3ea6424b',
```

Reply:
Ah ok, sure thing